### DISCUSSION TITLE
Built-in focus handling in VirtualizedList

### DISCUSSION THREAD

**Discussion by yusufyildirim** [react-native-tvos/react-native-tvos#665] [Feature Requests]
Managing focus in a list is pretty painful experience for developers. They first have to handle scroll position with `initialScrollIndex` and `contentOffset` so that the target element end up rendered in the viewport, which makes it focusable. After it's done, they somehow move focus to target element with either using `hasTVPreferredFocus` by leaking the responsibility to the rendered item or dealing with refs and using some sort of imperative method like `requestTVFocus` etc.

This is super tedious and error prone. It's quite possible to automate with all the info we have in VirtualizedList. I'm proposing to introduce a declarative way for initial focus action and one imperative way for other use-cases to delegate all this hassle to VirtualizedList itself.

The declarative way for initial focus actions:
```tsx
<FlatList
  // ...other props
  initialFocusIndex={5}
/>
```

The Imperative way:
```ts
flatListRef.current.focusToIndex({
  // Basically the same api as `scrollToIndex`
  index: number;
  animated?: boolean;
  viewOffset?: number;
  viewPosition?: number;
})
```

They're basically the focus equilavent of `initialScrollIndex` and `scrollToIndex`. Both of these will be responsible of handling all the aspects including scrolling, rendering (making sure the element is rendered and focusable) and moving focus. They'll also work in combination of the other features I proposed in the #663 and #664.

The challenging part is moving focus by index to a component automagically without knowing the structure of it or having direct access to the `Pressable` underneath. I have a simple trick in mind I'm hoping that'll help me achieve the result I want.

**Comment by pigeonmal** [comment]
Please add this !! Focus a specific item in a list it's a pain
